src/parseTools.js

```diff
@@ -819,20 +819,26 @@ function calcFastOffset(ptr, pos, noNeedFirst) {
 
 function makeGetSlabs(ptr, type, allowMultiple, unsigned) {
   assert(type);
-  if (isPointerType(type)) type = 'i32'; // Hardcoded 32-bit
-  switch(type) {
-    case 'i1': case 'i8': return [unsigned ? 'HEAPU8' : 'HEAP8']; break;
-    case 'i16': return [unsigned ? 'HEAPU16' : 'HEAP16']; break;
+  if (isPointerType(type)) {
+    type = 'i32'; // Hardcoded 32-bit
+  }
+  switch (type) {
+    case 'i1':
+    case 'i8':
+      return [unsigned ? 'HEAPU8' : 'HEAP8'];
+    case 'i16':
+      return [unsigned ? 'HEAPU16' : 'HEAP16'];
     case '<4 x i32>':
-    case 'i32': case 'i64': return [unsigned ? 'HEAPU32' : 'HEAP32']; break;
-    case 'double': return ['HEAPF64'];
+    case 'i32':
+    case 'i64':
+      return [unsigned ? 'HEAPU32' : 'HEAP32'];
+    case 'double':
+      return ['HEAPF64'];
     case '<4 x float>':
-    case 'float': return ['HEAPF32'];
-    default: {
-      throw 'what, exactly, can we do for unknown types in TA2?! ' + [new Error().stack, ptr, type, allowMultiple, unsigned];
-    }
+    case 'float':
+      return ['HEAPF32'];
   }
-  return [];
+  throw 'what, exactly, can we do for unknown types in TA2?! ' + [new Error().stack, ptr, type, allowMultiple, unsigned];
```

Comment:
We should probably fix this error message too.   I think TA2 refers to USE_TYPED_ARRAYS which was removed in 2015.    

Do you know why this function used to return more than one thing?   Were there multiple memory views that we used to try to keep in sync somehow?

Reply:
Yes, as you said, this is super-old...

I believe we used to not have a single linear memory, but separate ones per type. This was one of the approaches experimented with in the very early days.

Today I think this can return a single item, and the error here can just be "internal error".